src/battle.ts

```diff
@@ -994,6 +996,8 @@ export interface ServerPokemon extends PokemonDetails, PokemonHealth {
 	pokeball: string;
 	/** false if the pokemon cannot gigantamax, otherwise a string containing the full name of its G-max move */
 	gigantamax: string | false;
+	teraType: string;
+	terastallized: boolean;
```

Comment:
I find it pretty confusing to have `terastallized` defined as a boolean here but in other contexts it's a string. Is it possible to rename one of them?

Reply:
I was thinking that as well, just thought it would be pointless to send the type twice for both properties when a mon is terastallized. I could just have the server send over the type instead of a boolean if that's preferred.

Reply:
The boolean could be sent from the server with a name other than "teraType" or "terastallized"? Or, have the `terastallized` that's an actual type name which is being repurposed from `teraType` renamed to "terastallizedType" or something.

Reply:
I went with just having `terastallized` be a string that contains the tera type if the mon is terastallized (and an empty string if it isn't)